origin-notifications/migrations/20181019004743-create-push-subscription.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('PushSubscriptions', {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER
+      },
+      // not unique!
+      endpoint: {
+        type: Sequelize.STRING
+      },
+      keys: {
+        type: Sequelize.HSTORE
+      },
+      expirationTime: {
+        type: Sequelize.DATE
+      },
+      // not unique!
+      account: {
+        type: Sequelize.STRING
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    })
+  },
```

Comment:
I think there should be an index on (account, endpoint) since you have a query doing a lookup based on that. Otherwise it would cause the query to do a full table scan (which is terrible for performance if we start having lot of rows in this table).

Adding a line like this would do:
.then(() => queryInterface.addIndex('PushSubscriptions', ['account', 'endpoint']))

Reply:
nice ! :)